tests/Unit/Models/FamilyModelTest.php

```diff
@@ -6,12 +6,128 @@
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no registrations, and that's fine.
+        $this->assertEquals(0, $family->registrations()->count());
+
+        // Associate with two registrations
+        $registration1 = new Registration();
+        $registration1->family_id = $family->id;
+        $registration1->centre_id = factory(Centre::class)->create()->id;
+        $registration1->eligibility = "other";
+        $registration1->save();
+
+        $registration2 = new Registration();
+        $registration2->family_id = $family->id;
+        $registration2->centre_id = factory(Centre::class)->create()->id;
+        $registration2->eligibility = "other";
+        $registration2->save();
+
+        $registrations = [$registration1, $registration2];
+
+        // Check they're there.
+        $this->assertEquals(2, $family->registrations->count());
+        foreach ($family->registrations as $index => $registration) {
+            $this->assertEquals($registration->id, $registrations[$index]->id);
+        }
+    }
+
+        /** @test */
+    public function itCanHaveCarers()
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no carers, and that's fine.
+        $this->assertEquals(0, $family->carers()->count());
+
+        // Add 3 Carers
+        $carers = factory(Carer::class, 3)->make();
+        $family->carers()->saveMany($carers);
+
+        // Check they're there.
+        $this->assertEquals(3, $family->carers->count());
+        foreach ($family->carers as $index => $carer) {
+            $this->assertEquals($carer->id, $carers[$index]->id);
+        }
+    }
+
+    /** @test */
+    public function itCanHaveChildren()
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no kids, and that's fine.
+        $this->assertEquals(0, $family->children()->count());
+
+        // Add 3 Children
+        $children = factory(Child::class, 3)->make();
+        $family->carers()->saveMany($children);
+
+        // Check they're there.
+        $this->assertEquals(3, $family->children->count());
+        foreach ($family->children as $index => $child) {
+            $this->assertEquals($child->id, $children[$index]->id);
+        }
+    }
+
+    /** @test */
+    public function itCanAppendItsPrimaryCarerName()
+    {
+        // Make a family with carers
+        $family = factory(Family::class)->create();
+
+        // There is no pri_carer on a normal family
+        $this->assertArrayNotHasKey('pri_carer', $family->getAttributes());
+
+        // We can call a scoped family.
+        $pri_carer_family = Family::withPrimaryCarer()->find(1);
+
+        $attribs =  $pri_carer_family->getAttributes();
+        // There is a pri_carer attribute on a scoped family.
+        $this->assertArrayHasKey('pri_carer', $attribs);
```

Comment:
Seems odd to be in scope if it is empty. Is this desired?

Reply:
It can be empty, the family might not have a primary carer (it's technically a possible thing). However the scopeWithPrimartyCarer hoists the first carer int the list into a pir_carer attribute. If none exists you get null.